drivers/rtc/Kconfig.stm32_rtc

```diff
@@ -0,0 +1,18 @@
+config RTC_STM32
+	bool "STM32 RTC Driver"
+	depends on SOC_FAMILY_STM32
+	default n
+	select USE_STM32_HAL_RTC
+	select USE_STM32_HAL_RTC_EX
+	select USE_STM32_HAL_RCC_EX
+	help
+	  Build STM32 RTC driver.
+
+config RTC_STM32_API_REENTRANCY
```

Comment:
get rid of that option, just make your rtc drives api function always reentrant.

Reply:
Before doing so, what is the use case you you target here?
If there are more than one client and they require to set different configuration, this is already an issue.

Reply:
We wanted to mimic other RTC drivers such as https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/rtc/Kconfig.qmsi#L10

In case of our Zephyr application we do not need `RTC_STM32_API_REENTRANCY`

Reply:
I removed `RTC_STM32_API_REENTRANCY` option